test/e2e/trafficroute/universal_multizone/traffic_route.go

```diff
@@ -0,0 +1,295 @@
+			currentPercent2 := echo2Resp / float64(i+1) * 100
+
+			if within(currentPercent1, v1Weight-errorPercent, v1Weight+errorPercent) && within(currentPercent2, v2Weight-errorPercent, v2Weight+errorPercent) {
+				return nil
+			}
+		}
+		return errors.Errorf("resulted split values %v and %v aren't within original values %v and %v",
+			echo1Resp/float64(maxAttempts)*100, echo2Resp/float64(maxAttempts)*100, v1Weight, v2Weight)
+	}
+
+	It("should access all instances of the service", func() {
+		const trafficRoute = `
+type: TrafficRoute
+name: route-dc-to-echo
+mesh: default
+sources:
+  - match:
+      kuma.io/service: demo-client
+destinations:
+  - match:
+      kuma.io/service: echo-server_kuma-test_svc_8080
+conf:
+  loadBalancer:
+    roundRobin: {}
+  split:
+    - weight: 1
+      destination:
+        kuma.io/service: echo-server_kuma-test_svc_8080
+        version: v1
+    - weight: 1
+      destination:
+        kuma.io/service: echo-server_kuma-test_svc_8080
+        version: v2
+    - weight: 1
+      destination:
+        kuma.io/service: echo-server_kuma-test_svc_8080
+        version: v4
+`
+		Expect(NewClusterSetup().Install(YamlUniversal(trafficRoute)).Setup(global)).To(Succeed())
+
+		collectResponses("echo-server_kuma-test_svc_8080.mesh", []string{"echo-v1", "echo-v2", "echo-v4"}, []string{"echo-v3"})
+	})
+
+	It("should route 100 percent of the traffic to the different service", func() {
+		const trafficRoute = `
+type: TrafficRoute
+name: route-dc-to-echo
+mesh: default
+sources:
+  - match:
+      kuma.io/service: demo-client
+destinations:
+  - match:
+      kuma.io/service: echo-server_kuma-test_svc_8080
+conf:
+  loadBalancer:
+    roundRobin: {}
+  split:
+    - weight: 100
+      destination:
+        kuma.io/service: backend
+`
+		Expect(NewClusterSetup().Install(YamlUniversal(trafficRoute)).Setup(global)).To(Succeed())
+
+		collectResponses("backend.mesh", []string{"backend-v1"}, nil)
+		collectResponses("backend.mesh", []string{"backend-v1"}, []string{"echo-v1", "echo-v2", "echo-v3", "echo-v4"})
+	})
+
+	It("should route split traffic between the versions with 20/80 ratio", func() {
+		v1Weight := 80
+		v2Weight := 20
+
+		trafficRoute := fmt.Sprintf(`
+type: TrafficRoute
+name: route-dc-to-echo
+mesh: default
+sources:
+  - match:
+      kuma.io/service: demo-client
+destinations:
+  - match:
+      kuma.io/service: echo-server_kuma-test_svc_8080
+conf:
+  loadBalancer:
+    roundRobin: {}
+  split:
+    - weight: %d
+      destination:
+        kuma.io/service: echo-server_kuma-test_svc_8080
+        version: v1
+    - weight: %d
+      destination:
+        kuma.io/service: echo-server_kuma-test_svc_8080
+        version: v2
+`, v1Weight, v2Weight)
+		Expect(NewClusterSetup().Install(YamlUniversal(trafficRoute)).Setup(global)).To(Succeed())
+
+		err := waitRatio("echo-server_kuma-test_svc_8080.mesh", v1Weight, v2Weight, "echo-v1", "echo-v2")
+		Expect(err).ToNot(HaveOccurred())
+	})
+}
```

Comment:
I feel like `waitRatio` and `collectResponses` are a bit cryptic. Names do not suggest that those are assertion and I think both of them contains a bit complicated logic.

Here's what I would do:
* Write a method `collectResponses := func(destination string) map[string]int` (instance to responses) that only sends 10 (can be 100 if needed) requests and collects responses, don't do any assertions
* In `should access all instances of the service` I would write - eventually (collectResponses and then do assertion in the test that there is 1,2,4 and not 3)
* In `should route 100 percent of the traffic to the different service` I would write - eventually (collectResponses and then do assertion that responses are from one instance)
* In `should route split traffic between the versions with 20/80 ratio` I would write - eventually (collectResponses and count if the traffic is propagated properly)

I think it will be also more flexible to add new test cases. Also I think `collectResponses` could be reused between both tests.